platform/ios/Integration Tests/MGLNetworkConfigurationIntegrationTests.mm

```diff
@@ -47,38 +54,214 @@ - (void)test1_NativeNetworkManagerDelegateIsSet
 
 - (void)test2_NativeNetworkManagerDelegateIsSetBySharedManager
 {
-    ASSERT_DELEGATE_IS_NIL();
+    ASSERT_NATIVE_DELEGATE_IS_NIL();
 
     // Just calling the shared manager is also sufficient (even though, it's a
     // singleton and created with a dispatch_once, the delegate is re-set for
     // each call.
     [MGLNetworkConfiguration sharedManager];
-    ASSERT_DELEGATE_IS_NOT_NIL();
+    ASSERT_NATIVE_DELEGATE_IS_NOT_NIL();
 }
 
 - (void)test3_NativeNetworkManagerDelegateIsSetBySharedOfflineStorage
 {
-    ASSERT_DELEGATE_IS_NIL();
+    ASSERT_NATIVE_DELEGATE_IS_NIL();
 
     // Similar to `[MGLNetworkConfiguration sharedManager]`,
     // `[MGLOfflineStorage sharedOfflineStorage]` also sets the delegate.
     [MGLOfflineStorage sharedOfflineStorage];
-    ASSERT_DELEGATE_IS_NOT_NIL();
+    ASSERT_NATIVE_DELEGATE_IS_NOT_NIL();
 }
 
 - (void)test4_NativeNetworkManagerDelegateIsSetBySharedOfflineStorageASecondTime
 {
     // Testing a second time...
-    ASSERT_DELEGATE_IS_NIL();
+    ASSERT_NATIVE_DELEGATE_IS_NIL();
     [MGLOfflineStorage sharedOfflineStorage];
-    ASSERT_DELEGATE_IS_NOT_NIL();
+    ASSERT_NATIVE_DELEGATE_IS_NOT_NIL();
 }
 
 - (void)test5_NativeNetworkManagerDelegateIsSetByMapViewInit
 {
-    ASSERT_DELEGATE_IS_NIL();
+    ASSERT_NATIVE_DELEGATE_IS_NIL();
     (void)[[MGLMapView alloc] init];
-    ASSERT_DELEGATE_IS_NOT_NIL();
+    ASSERT_NATIVE_DELEGATE_IS_NOT_NIL();
+}
+
+- (void)testNetworkConfigurationDelegateIsNil
+{
+    MGLNetworkConfiguration *manager = [MGLNetworkConfiguration sharedManager];
+    XCTAssertNil(manager.delegate);
```

Comment:
Would it be possible to convert these integration tests into unit tests? At a glance, they don’t look too different than the existing unit tests of MGLOfflineStorage.

Reply:
Yes, I think so; let's do that in another PR.